### PR TITLE
ch4/ofi: fix counter increment in MPIDI_OFI_win_trigger_rma_progress

### PR DESCRIPTION
**Pull request by abrooks98**
## Pull Request Description

Progress counter should be increased by 1 before checking with interval the limit. Otherwise the first operation in a window always triggers flushing.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by abrooks98** [comment]
test:mpich/ch4/ofi

File: src/mpid/ch4/netmod/ofi/ofi_win.h [hunk 1]
```diff
@@ -98,11 +98,11 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_win_trigger_rma_progress(MPIR_Win * win,
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_WIN_TRIGGER_RMA_PROGRESS);
 
     if (!MPIDI_OFI_ENABLE_DATA_AUTO_PROGRESS && MPIR_CVAR_CH4_OFI_RMA_PROGRESS_INTERVAL != -1) {
+        MPIDI_OFI_WIN(win).progress_counter++;
         if (MPIDI_OFI_WIN(win).progress_counter % MPIR_CVAR_CH4_OFI_RMA_PROGRESS_INTERVAL == 0) {
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_win.h:102]: this is unrelated, but if we replace the modulo operator with `>` or `>=` it should reduce the cost of this check by a few instructions.

Comment (raffenet) [reply src/mpid/ch4/netmod/ofi/ofi_win.h:102]: compiler explorer example (godbolt.org):
```
num % 100 == 0
        mov     edx, DWORD PTR [rbp-4]
        movsx   rax, edx
        imul    rax, rax, 1374389535
        shr     rax, 32
        sar     eax, 5
        mov     ecx, edx
        sar     ecx, 31
        sub     eax, ecx
        imul    ecx, eax, 100
        mov     eax, edx
        sub     eax, ecx
        test    eax, eax
        sete    al
```
```
num >= 0
        cmp     DWORD PTR [rbp-4], 99
        setg    al
```

**Comment by abrooks98** [comment]
test:mpich/ch4/ofi

**Review by raffenet — APPROVED** [review]
LGTM

**Pull request merged**